packages/@expo/metro-config/src/transform-worker/__tests__/metro-transform-worker.test.ts

```diff
@@ -121,24 +154,75 @@ it('transforms a simple script', async () => {
       "})(typeof globalThis !== 'undefined' ? globalThis : typeof global !== 'undefined' ? global : typeof window !== 'undefined' ? window : this);",
     ].join('\n')
   );
-  expect(result.output[0].data.map).toMatchSnapshot();
-  expect(result.output[0].data.functionMap).toMatchSnapshot();
+
+  const trace = toTraceMap(result.output[0], contents);
+
+  expect(generatedPositionFor(trace, { source: '', line: 1, column: 0 })).toMatchObject({
+    line: 2,
+    column: 2,
+  });
+
+  expect(originalPositionFor(trace, { line: 2, column: 2 })).toMatchObject({
+    line: 1,
+    column: 0,
+    name: 'someReallyArbitrary',
+  });
+
+  // NOTE: If downgraded below @babel/generator@7.21.0, names will be missing here
+  expect(originalPositionFor(trace, { line: 3, column: 10 })).toMatchObject({
+    line: 1,
+    column: 25,
+    name: 'globalThis',
+  });
+  expect(originalPositionFor(trace, { line: 3, column: 59 })).toMatchObject({
+    line: 1,
+    column: 25,
+    name: 'global',
+  });
+  expect(originalPositionFor(trace, { line: 3, column: 100 })).toMatchObject({
+    line: 1,
+    column: 25,
+    name: 'window',
+  });
+
   expect(result.dependencies).toEqual([]);
 });
 
 it('transforms a simple module', async () => {
+  const contents = 'arbitrary(code)';
+
   const result = await Transformer.transform(
     baseConfig,
     '/root',
     'local/file.js',
-    Buffer.from('arbitrary(code)', 'utf8'),
+    Buffer.from(contents, 'utf8'),
     baseTransformOptions
   );
 
+  const trace = toTraceMap(result.output[0], contents);
+
+  expect(generatedPositionFor(trace, { source: '', line: 1, column: 0 })).toMatchObject({
+    line: 2,
+    column: 2,
+  });
+  expect(generatedPositionFor(trace, { source: '', line: 1, column: 10 })).toMatchObject({
+    line: 2,
+    column: 12,
+  });
+
+  expect(originalPositionFor(trace, { line: 2, column: 2 })).toMatchObject({
+    line: 1,
+    column: 0,
+    name: 'arbitrary',
+  });
+  expect(originalPositionFor(trace, { line: 2, column: 12 })).toMatchObject({
+    line: 1,
+    column: 10,
+    name: 'code',
+  });
+
   expect(result.output[0].type).toBe('js/module');
   expect(result.output[0].data.code).toBe([HEADER_DEV, '  arbitrary(code);', '});'].join('\n'));
-  expect(result.output[0].data.map).toMatchSnapshot();
-  expect(result.output[0].data.functionMap).toMatchSnapshot();
```

Comment:
`functionMap` is a facebook thing which maps to the nonstandard `x_facebook_sources`, imo we should keep the test check since I have no idea how it's intended to be used. https://metrobundler.dev/docs/source-map-format/#x_facebook_sources